kivy/tests/test_uix_settings.py

```diff
@@ -31,3 +33,73 @@ def test_settings_create_json_panel_errors():
     ):
         data = json.dumps([{"type": "testunknown"}])
         Settings().create_json_panel("Demo", config, filename=None, data=data)
+
+
+class SettingNumericTest(SettingNumeric):
+
+    def __init__(self, panel):
+        print("SettingNumericTest")
+        self.panel = panel
+        self.section = "Section"
+        self.key = "Key"
+        self.value = "0"
+        self.textinput = TextInput(
+            text=self.value, font_size='24sp', multiline=False,
+            size_hint_y=None, height='42sp'
+        )
+        super(SettingNumericTest, self).__init__()
+
+
+class TestSettingNumeric(unittest.TestCase):
+
+    @pytest.fixture(autouse=True, scope="class")
+    def setup_once(self, request):
+        config = ConfigParser()
+        config.setdefaults('Section', {'Key': 0})
+
+        request.cls.panel = Settings().create_json_panel("TestSettingNumeric", config, filename=None, data="""
+            [
+                {
+                    "type": "title", 
+                    "title": "Title"
+                },
+                {
+                    "type": "numeric", 
+                    "title": "Numerical value",
+                    "desc": "Set a mumeric value",
+                    "section": "Section", 
+                    "key": "Key"
+                }
+            ]"""
+        )
+
+
+    @pytest.fixture(autouse=True, scope="function")
+    def setup(self):
+        self.settingNumeric = SettingNumericTest(self.panel)
```

Comment:
I am concerned here. You are actually testing a subclass of `SettingNumeric`, not an instance of `SettingNumeric` itself. You've added a bunch of properties which could interfere with it's behaviour. Even if not now, it could in future versions. Rather build the test to test a pure `SettingNumeric` instance.